### ISSUE TITLE
Duplicate entries in fishfile

### ISSUE THREAD

**Issue by mrak** [fisherman/fisherman#371]
I use my own set of fish scripts as a plugin, so I have this in my `fishfile`:

```
mrak/mrak.fish
```

For convenience, I have set the origin push url to be `git@github.com:mrak/mrak.fish` so that I can my local edits and push them. I have also tried, instead of modifying the origin push url, adding a new remote as `upstream`

When I run `fisher up`, it adds the following to my fishfile:

```
mrak/mrak.fish
mrak/mrak.fish
git@github.com:mrak/mrak.fish
```

```
$ git -C ~/.config/fisherman/mrak.fish remote -v
origin	https://github.com/mrak/mrak.fish (fetch)
origin	https://github.com/mrak/mrak.fish (push)
upstream	git@github.com:mrak/mrak.fish.git (fetch)
upstream	git@github.com:mrak/mrak.fish.git (push)
```

It would be nice if the first `url` entry from `.git/config` is used instead of _all_ of them :)

**Comment by jbucaran** [comment]
@mrak Oh, I vaguely remember seeing something like this, but I'm not sure. What version are you running?

**Comment by mrak** [comment]
```
$ fisher --version
fisherman version 2.12.0 ~/.config/fish/functions/fisher.fish
```

**Comment by jbucaran** [comment]
@mrak What's mrak/mrak.fish by the way? Don't you mean only `mrak`?

**Comment by jbucaran** [comment]
@mrak It seems fisherman is doing the wrong thing, because your fishfile should not have any duplicates, on the other hand `mrak/mrak.fish` seems like a bad name, since fish already uses the `.fish` extension for script files, but `mrak/mrak.fish` is actually a directory.

I think @laughedelic was having the same issue?

**Comment by jbucaran** [comment]
Maybe related?
* https://github.com/fisherman/fisherman/issues/242

**Comment by mrak** [comment]
ah, `mrak/mrak.fish` is just my github repo with my fish configuration: https://github.com/mrak/mrak.fish

**Comment by jbucaran** [comment]
@mrak Could you confirm this behavior persists if you rename mrak.fish to not end in .fish?

**Comment by mrak** [comment]
I created a simple fish plugin at https://github.com/mrak/fisherman_371, but you could the following with any github fish plugin you create.

```
$ fisher mrak/fisherman_371
$ cd ~/.config/fisherman/fisherman_371
$ git remote add upstream git@github.com:mrak/fisherman_371
$ git remote -v
origin  https://github.com/mrak/fisherman_371 (fetch)
origin  https://github.com/mrak/fisherman_371 (push)
upstream        git@github.com:mrak/fisherman_371 (fetch)
upstream        git@github.com:mrak/fisherman_371 (push)
$ fisher up
$ cat ~/.config/fish/fishfile
mrak/fisherman_371
git@github.com:mrak/fisherman_371
$ cd ~/.config/fisherman/fisherman_371
$ git remote set-url --push origin git@github.com:mrak/fisherman_371
$ git remote -v
origin  https://github.com/mrak/fisherman_371 (fetch)
origin  git@github.com:mrak/fisherman_371 (push)
upstream        git@github.com:mrak/fisherman_371 (fetch)
upstream        git@github.com:mrak/fisherman_371 (push)
$ fisher up
$ cat ~/.config/fish/fishfile
mrak/fisherman_371
git@github.com:mrak/fisherman_371
git@github.com:mrak/fisherman_371

**Comment by jbucaran** [comment]
Thanks a lot @mrak!

I think a good place to start investigating is here: 
* https://github.com/fisherman/fisherman/blob/master/fisher.fish#L1640

We're reading .git/config and parsing the URL here:
* https://github.com/fisherman/fisherman/blob/master/fisher.fish#L1651

Perhaps we need to change this regex `/https:\/\/gist/`, or use git to query the right info.

Would you like to hack on this?

**Comment by mrak** [comment]
I can take a look. I'm pretty sure it's the `awk` script iterating over all `url` lines in the `.git/config` instead of just finding the one for `origin`.

Is there a way to prevent fisherman from updating itself every time you run `fisher up`? It always blows away any local changes I do

**Comment by jbucaran** [comment]
@mrak There is no conventional way. I simply comment it out during local development, but I am open to more sophisticated workflows!

I think installing fisherman _with_ fisherman locally could work.

> I can take a look. I'm pretty sure it's the awk script iterating over all url lines in the .git/config instead of just finding the one for origin.

Please do 🙏. I noticed you were not in the org, so I sent you an invite.

**Comment by jbucaran** [comment]
@aristidesfl Could you have a look as well?

**Comment by aristidesfl** [comment]
I've been just using `source fisher.fish` when working on plugins. Is there an even simpler way?

**Comment by jbucaran** [comment]
@mrak Does that work for you?

**Comment by mrak** [comment]
for now I'm using `fisher up <plugin name>` which appears to skip the self-update

**Comment by jbucaran** [comment]
@mrak Ah, yup, I forgot about that one. 👍

**Comment by mrak** [comment]
addressed by #372
